python_modules/libraries/dagster-datadog/dagster_datadog/resources.py

```diff
@@ -0,0 +1,58 @@
+from datadog import initialize, statsd, DogStatsd
+
+from dagster import resource, Dict, Field, String
+
+
+class DataDogResource:
+    '''DataDogResource
+
+    This resource is a thin wrapper over the dogstatsd library:
+
+    https://datadogpy.readthedocs.io/en/latest/#datadog-dogstatsd-module
+
+    As such, we directly mirror the public API methods of DogStatsd here; you can refer to the
+    DataDog documentation above for how to use this resource.
+    '''
+
+    # Mirroring levels from the dogstatsd library
+    OK, WARNING, CRITICAL, UNKNOWN = (
+        DogStatsd.OK,
+        DogStatsd.WARNING,
+        DogStatsd.CRITICAL,
+        DogStatsd.UNKNOWN,
+    )
+
+    def __init__(self, api_key, app_key):
+        initialize(api_key=api_key, app_key=app_key)
+
+        # Pull in methods from the dogstatsd library
+        for method in [
+            'event',
+            'gauge',
+            'increment',
+            'decrement',
+            'histogram',
+            'distribution',
+            'set',
+            'service_check',
+            'timed',
+            'timing',
+        ]:
+            setattr(self, method, getattr(statsd, method))
```

Comment:
yeah you can do this.

the other option is to override ``__getattr__`` and do this dynamically to future proof against future additions to the public API

Reply:
yeah, I wanted to avoid a wholesale grab since they didn't consistently use underscore prefixes on their private methods, so we'd pull over stuff that we probably shouldn't